include/zephyr/modem/cmux.h

```diff
@@ -195,7 +195,7 @@ void modem_cmux_init(struct modem_cmux *cmux, const struct modem_cmux_config *co
 struct modem_cmux_dlci_config {
 	/** DLCI channel address */
 	uint8_t dlci_address;
-	/** Receive buffer used by pipe */
+	/** Receive buffer used by pipe; must not be reused anywhere. */
```

Comment:
question: there are a few `receive_buf` in this file, why only mention `"must not be reused anywhere."` here? is there any receive buffer that can be reused and this one is an exception?

Reply:
No, there is no receive buffer that can be reused. I added the comment here because I managed to trap myself with this buffer, so I though I'd add a warning to it.
When using CMUX, the DLCIs kind of are the point of communication between the user and the CMUX module. Not so much the other buffers.
Though anyway the most important place is `modem_pipe_receive()`, as that's how all the receiving takes place.

Reply:
> No, there is no receive buffer that can be reused.

In that case wouldn't this exception be rather confusing? i.e. now in the documentation you will have receive buffers and receive buffer that must not be reused anywhere

Reply:
Could be. Although no buffer is specified as being reusable, but I get your point.
I could change the comment to `make sure not to reuse it anywhere`; what do you think?

Reply:
IMO this is simply about consistency, the fact that you only change this line is pretty random - you happened to have mistakenly shared a buffer for this type of struct which caused you some troubles - it probably could have been any other structs.

I'm not sure if anyone else would assume that a buffer can be shared among this kind of config struct, but one might start to wonder if (s)he saw that just one struct noted that its `receive_buf` shouldn't be reused, and the buffer of the same nature exists in a few other structs in this header.

tl;dr: probably not change this line, or change all of the lines, for consistency?

Reply:
Alright. Removed it.